### PR TITLE
ascent: add version v0.6.0

### PR DESCRIPTION
**Pull request by eugeneswalker**
Ascent: add version `0.6.0` and make it the preferred version.

@cyrush

### CODE REVIEW

**Comment by eugeneswalker** [comment]
Should we make `v0.6.0` the preferred version?

I've been seeing `ascent@develop` fail in all my pipelines for perhaps at least a couple weeks.

File: var/spack/repos/builtin/packages/ascent/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ascent/package.py]: Versioned versions (vs. branches like `develop`) are usually listed by `tag` (i.e., `tag='v0.6.0'`).

I see a tag in the repo for this version but I'm not seeing a branch with that label.

Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/ascent/package.py]: Good catch! Interestingly, I tried this and it worked ?? Either way, I will change to tag. Thanks!

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/ascent/package.py]: I'm not surprised that you could install the version -- through `git` perhaps -- but I wonder if it would've picked up the `submodules=True` option.

**Review by tldahlgren — DISMISSED** [review]
LGTM.  Approval contingent on all required checks passing.

Would also like to hear from Cyrus before the package is merged.

**Comment by eugeneswalker** [comment]
I am noticing that even with the addition of a numeric version, `ascent` still concretizes to `ascent@develop` by default. Is it because `0.6.0` is from a tag where `develop` is from a branch?

**Comment by cyrush** [comment]
Thanks. I started this process in nov, but was stalled with a few dependent packages. Those have now been merged. 

For the current state of spack develop, I think the stars align such that v0.6.0 will build and ascent@develop has diverged. 

Our goal has been to keep develop always healthy, but delays in updating other packages undermined that a bit.

I am fine with switching to 0.6.0 as the default if thats the one that builds :-). I won't have time to make develop healthy in the immediate future.

**Comment by eugeneswalker** [comment]
ill make 0.6.0 preferred then. thank you for the help!

**Comment by eugeneswalker** [comment]
> LGTM. Approval contingent on all required checks passing.
> 
> Would also like to hear from Cyrus before the package is merged.

@tldahlgren Following Cyrus's post, I modified this PR to make 0.6.0 preferred. As such I need a re-approval before it will let me merge

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**
